src/fu-util.c

```diff
@@ -377,56 +388,75 @@ fu_util_perhaps_show_unreported (FuUtilPrivate *priv, GError **error)
-			unreported_success++;
+			g_ptr_array_add (devices_success, dev);
 			break;
 		default:
 			break;
 		}
 	}
 
 	/* nothing to do */
-	if (unreported_failed == 0 && unreported_success == 0)
+	if (devices_failed->len == 0 && devices_success->len == 0) {
+		g_debug ("no unreported devices");
 		return TRUE;
+	}
 
-	/* nag the user to do something that requires opt-in */
-	g_printerr ("\n******************\n\n");
-	if (unreported_failed > 0) {
-		g_printerr ("%s: %s\n",
-			    /* TRANSLATORS: we failed to apply a firmware update */
-			    _("WARNING"),
-			    /* TRANSLATORS: explain why we want to upload */
-			    ngettext ("A firmware update failed to be applied",
-				      "Firmware updates failed to be applied",
-				      unreported_failed));
-	} else if (unreported_success > 0) {
-		g_printerr ("%s: %s\n",
-			    /* TRANSLATORS: we did a firmware update well */
-			    _("INFO"),
-			    /* TRANSLATORS: explain why we want to upload */
-			    ngettext ("A firmware update was applied successfully",
-				      "Firmware updates were applied successfully",
-				      unreported_success));
-	}
-	g_printerr ("%s\n > fwupdmgr report-history\n",
-		    /* TRANSLATORS: what the user has to do, command follows */
-		    _("To share useful information with the developers and "
-		      "clear this message use the following command:"));
-	g_printerr ("\n******************\n");
-	return TRUE;
+	/* show the success and failures */
+	if (!priv->assume_yes) {
+
+		/* delimit */
+		g_print ("________________________________________________\n");
+
+		/* failures */
+		if (devices_failed->len > 0) {
+			/* TRANSLATORS: a list of failed updates */
+			g_print ("\n%s\n\n", _("Devices that were not updated correctly:"));
+			for (guint i = 0; i < devices_failed->len; i++) {
+				FwupdDevice *dev = g_ptr_array_index (devices_failed, i);
+				FwupdRelease *rel = fwupd_device_get_release_default (dev);
+				g_print (" • %s (%s → %s)\n",
+					 fwupd_device_get_name (dev),
+					 fwupd_device_get_version (dev),
+					 fwupd_release_get_version (rel));
+			}
+		}
+
+		/* success */
+		if (devices_success->len > 0) {
+			/* TRANSLATORS: a list of successful updates */
+			g_print ("\n%s\n\n", _("Devices that have been updated successfully:"));
+			for (guint i = 0; i < devices_success->len; i++) {
+				FwupdDevice *dev = g_ptr_array_index (devices_success, i);
+				FwupdRelease *rel = fwupd_device_get_release_default (dev);
+				g_print (" • %s (%s → %s)\n",
+					 fwupd_device_get_name (dev),
+					 fwupd_device_get_version (dev),
+					 fwupd_release_get_version (rel));
+			}
+		}
+
+		/* ask for permission */
+		g_print ("\n%s (%s) [Y|n]: ",
+			 /* TRANSLATORS: explain why we want to upload */
+			 _("Upload report now?"),
+			 /* TRANSLATORS: metadata is downloaded from the Internet */
+			 _("Requires internet connection"));
+		if (!fu_util_prompt_for_boolean (TRUE))
+			return TRUE;
+	}
+
+	/* this still asks for user confirmation */
```

Comment:
This comment seems wrong.  if priv->assume_yes was true they would get down here with no confirmation.

Reply:
Nahh, they'd get asked in `fu_util_prompt_for_boolean()` in `fu_util_report_history_for_uri` -- maybe that's a bug in itself. If the user specifies -y then maybe we should just takes that as "yes, always"

Reply:
Ah yeah, that's what I would expect.